### PR TITLE
Fix #13046: Simplify block sectioned comparisons

### LINKED ISSUE #13046: Simplify block sectioned comparisons
Lots of places in the code use the following comparison:
https://github.com/OpenRCT2/OpenRCT2/blob/47c72eaed3d9a14303dee5baa78f5772b63cd215/src/openrct2-ui/windows/Ride.cpp#L3826

But there is a simpler version which is a lot easier to read:
```c++
ride->IsBlockSectioned()
```

Please replace all the places in the code using the former for the latter

### PR DESCRIPTION
**Pull request by SGunner2014**

### CODE REVIEW

**Review by tupaschoal — COMMENTED** [review]
You're on the right path, but I think you've missed some. From a quick count, I'd say there are:
- 5 candidates on  ` src/openrct2-ui/windows/Ride.cpp`
- 7 on `src/openrct2/ride/Ride.cpp`
- 3 on `src/openrct2/ride/Vehicle.cpp`

File: src/openrct2/ride/Ride.cpp [hunk 1]
```diff
@@ -4936,7 +4936,7 @@ static bool ride_create_cable_lift(ride_id_t rideIndex, bool isApplying)
     if (ride == nullptr)
         return false;
 
-    if (ride->mode != RideMode::ContinuousCircuitBlockSectioned && ride->mode != RideMode::ContinuousCircuit)
+    if (!(ride->IsBlockSectioned()))
```
Comment (AaronVanGeffen) [inline src/openrct2/ride/Ride.cpp]: This statement contains superfluous parentheses. I suggest:
```suggestion
    if (!ride->IsBlockSectioned())
```

File: src/openrct2/ride/Vehicle.cpp [hunk 2]
```diff
@@ -3431,8 +3431,7 @@ void Vehicle::FinishDeparting()
         audio_play_sound_at_location(SoundId::RideLaunch1, { x, y, z });
     }
 
-    if (curRide->mode != RideMode::Race && curRide->mode != RideMode::ContinuousCircuitBlockSectioned
-        && curRide->mode != RideMode::PoweredLaunchBlockSectioned)
+    if (curRide->mode != RideMode::Race && !(curRide->IsBlockSectioned()))
```
Comment (tupaschoal) [inline src/openrct2/ride/Vehicle.cpp:3434]:
```suggestion
    if (curRide->mode != RideMode::Race && !curRide->IsBlockSectioned())
```
Remove these parentheses as well.

One tip, when you fix the issue we open, make sure you set the conversation as resolved.

**Review by tupaschoal — COMMENTED** [review]
This looks almost ready to go now, just need to remove some parentheses

File: src/openrct2/ride/Vehicle.cpp (hunk 2 above)
Comment (SGunner2014) [reply src/openrct2/ride/Vehicle.cpp:3434]: Okay, will do, thanks

**Review by tupaschoal — APPROVED** [review]

**Pull request merged**